--- a/kernel/apk_sign.c
+++ b/kernel/apk_sign.c
@@ -361,5 +361,5 @@ bool is_manager_apk(char *path)
 #endif
 	return check_v2_signature(path, EXPECTED_MANAGER_SIZE, EXPECTED_MANAGER_HASH)
-	|| check_v2_signature(path, 0x3e6, "79e590113c4c4c0c222978e413a5faa801666957b1212a328e46c00c69821bf7"); //KernelSU-Next
+	|| check_v2_signature(path, 0x3e6, "79e590113c4c4c0c222978e413a5faa801666957b1212a328e46c00c69821bf7") //KernelSU-Next
+	|| check_v2_signature(path, 0x338, "f26471a28031130362bce7eebffb9a0b8afc3095f163ce0c75a309f03b644a1f"); // pershoot Manager
 }
